internal/core/src/exec/expression/Expr.h

```diff
@@ -204,13 +220,16 @@ class SegmentExpr : public Expr {
 
     void
     MoveCursor() override {
-        if (is_index_mode_) {
-            MoveCursorForIndex();
-            if (segment_->HasFieldData(field_id_)) {
+        // when we specify input, do not maintain states
+        if (has_offset_input_) {
```

Comment:
Why it does not need to move cursor here when input is specified? Not the other way around as in `internal/core/src/exec/expression/LogicalBinaryExpr.h` per say?